app/src/main/java/br/com/dionataferraz/vendas/account/AcountViewModel.kt

```diff
@@ -1,49 +1,55 @@
 package br.com.dionataferraz.vendas.account
 
-import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
-import br.com.dionataferraz.vendas.account.data.Account
-import br.com.dionataferraz.vendas.account.data.local.AccountDatabase
-import br.com.dionataferraz.vendas.account.data.local.AccountEntity
-import br.com.dionataferraz.vendas.account.data.local.Operation
+import androidx.lifecycle.viewModelScope
 import br.com.dionataferraz.vendas.account.usecase.findAccountUseCase
 import br.com.dionataferraz.vendas.account.usecase.updateAccountUseCase
+import br.com.dionataferraz.vendas.transaction.TransactionType
+import br.com.dionataferraz.vendas.transaction.data.TransactionRequest
+import br.com.dionataferraz.vendas.transaction.usecase.InsertTransactionUsecase
+import kotlinx.coroutines.launch
 
 class AccountViewModel : ViewModel() {
 
     var error = MutableLiveData<String>()
     var success = MutableLiveData<Double>()
 
-    fun withdrawAcc(id: Int, amount: Double) {
+    fun registerItemBought(id: Int, value: Double, type: TransactionType) {
         try {
-            val acc = findAccountUseCase(id).invoke()
-            validateWithdraw(amount, acc.value)
-            updateAccountUseCase(
+            validateCredit(value)
+            val acc = findAccountUseCase().invoke(id)
+
+            val newTransaction = TransactionRequest(
+                value = value,
+                description = type.value,
+                transactionType = type
+            )
+
+            viewModelScope.launch() {
+                InsertTransactionUsecase(newTransaction).register()
```

Comment:
Neste caso poderia ter criado um` val by lazy` para o usecase, pq utilizar o `val by lazy`? fica mais facil na reutilização para as outras funções
```suggestion
                usecase.register(newTransaction)
```